File: ts/InstantDiscountSystem.ts
import * as readline from "readline";

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

const ask = (q: string): Promise<string> => new Promise(res => rl.question(q, res));

async function main() {
  console.log("Welcome, dear customer!");

  const input = await ask("Please enter the amount: ");
  const amount = parseFloat(input);

  if (amount <= 0) {
    console.log("The amount cannot be zero or negative!");
  } else {
    let discount = 0;
    if (amount >= 500) discount = amount * 0.2;
    else if (amount >= 100) discount = amount * 0.1;

    console.log(`Your available discount is: ${discount.toFixed(2)}`);
    console.log(`Your total amount after discount is: ${(amount - discount).toFixed(2)}`);
  }

  rl.close();
}

main();

File: rust/instant_discount_system.rs
use std::io::{self, Write};

fn main() {
    println!("Welcome, dear customer!");

    print!("Please enter the amount: ");
    io::stdout().flush().unwrap();

    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let amount: f64 = input.trim().parse().expect("Invalid number");

    if amount <= 0.0 {
        println!("The amount cannot be zero or negative!");
    } else {
        let discount = if amount >= 500.0 {
            amount * 0.2
        } else if amount >= 100.0 {
            amount * 0.1
        } else {
            0.0
        };

        println!("Your available discount is: {:.2}", discount);
        println!("Your total amount after discount is: {:.2}", amount - discount);
    }
}
